# 
# Submitted:
#
# Ladislav Michl, 2004-10-23
#
# Error: 
#
# The 2.6.x kernel removed '__kernel_dev_t' and renamed it as
# '__kernel_old_dev_t'.
#
# Description: 
#
# http://www.busybox.net/lists/busybox/2004-August/012322.html
# http://www.uclibc.org/lists/uclibc-cvs/2004-February/004301.html
#
# State:
# 
# Fixed in CVS version.

diff -Naur uClibc-0.9.26-orig/libc/sysdeps/linux/alpha/bits/kernel_types.h uClibc-0.9.26/libc/sysdeps/linux/alpha/bits/kernel_types.h
--- uClibc-0.9.26-orig/libc/sysdeps/linux/alpha/bits/kernel_types.h	2002-08-27 03:20:10.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/alpha/bits/kernel_types.h	2004-02-06 08:07:06.000000000 +0100
@@ -32,6 +32,7 @@
 typedef __kernel_gid_t __kernel_old_gid_t;
 typedef __kernel_uid_t __kernel_uid32_t;
 typedef __kernel_gid_t __kernel_gid32_t;
+typedef __kernel_dev_t __kernel_old_dev_t;
 
 typedef struct {
 	int val[2];
diff -Naur uClibc-0.9.26-orig/libc/sysdeps/linux/arm/bits/kernel_types.h uClibc-0.9.26/libc/sysdeps/linux/arm/bits/kernel_types.h
--- uClibc-0.9.26-orig/libc/sysdeps/linux/arm/bits/kernel_types.h	2002-08-27 03:20:11.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/arm/bits/kernel_types.h	2004-02-06 08:07:06.000000000 +0100
@@ -31,6 +31,7 @@
 typedef unsigned short		__kernel_old_uid_t;
 typedef unsigned short		__kernel_old_gid_t;
 typedef long long		__kernel_loff_t;
+typedef __kernel_dev_t		__kernel_old_dev_t;
 
 typedef struct {
 #ifdef __USE_ALL
diff -Naur uClibc-0.9.26-orig/libc/sysdeps/linux/cris/bits/kernel_types.h uClibc-0.9.26/libc/sysdeps/linux/cris/bits/kernel_types.h
--- uClibc-0.9.26-orig/libc/sysdeps/linux/cris/bits/kernel_types.h	2002-09-16 10:08:35.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/cris/bits/kernel_types.h	2004-02-06 08:09:46.000000000 +0100
@@ -7,19 +7,19 @@
 typedef unsigned long	__kernel_ino_t;
 typedef unsigned short	__kernel_mode_t;
 typedef unsigned short	__kernel_nlink_t;
-typedef long			__kernel_off_t;
-typedef int				__kernel_pid_t;
+typedef long		__kernel_off_t;
+typedef int		__kernel_pid_t;
 typedef unsigned short  __kernel_ipc_pid_t;
 typedef unsigned short	__kernel_uid_t;
 typedef unsigned short	__kernel_gid_t;
 typedef unsigned long	__kernel_size_t;
-typedef long			__kernel_ssize_t;
-typedef int				__kernel_ptrdiff_t;
-typedef long			__kernel_time_t;
+typedef long		__kernel_ssize_t;
+typedef int		__kernel_ptrdiff_t;
+typedef long		__kernel_time_t;
 typedef long            __kernel_suseconds_t;
-typedef long			__kernel_clock_t;
-typedef int				__kernel_daddr_t;
-typedef char *			__kernel_caddr_t;
+typedef long		__kernel_clock_t;
+typedef int		__kernel_daddr_t;
+typedef char *		__kernel_caddr_t;
 typedef unsigned short  __kernel_uid16_t;
 typedef unsigned short  __kernel_gid16_t;
 typedef unsigned int    __kernel_uid32_t;
@@ -27,9 +27,10 @@
 
 typedef unsigned short  __kernel_old_uid_t;
 typedef unsigned short  __kernel_old_gid_t;
+typedef __kernel_dev_t	__kernel_old_dev_t;
 
 #ifdef __GNUC__
-typedef long long		__kernel_loff_t;
+typedef long long	__kernel_loff_t;
 #endif
 
 typedef struct {
@@ -38,7 +39,7 @@
 #else
 	int	__val[2];
 #endif
-} __kernel_fsid_t;
+    } __kernel_fsid_t;
 
 /* should this ifdef be here ?  */

diff -Naur uClibc-0.9.26-orig/libc/sysdeps/linux/e1/bits/kernel_types.h uClibc-0.9.26/libc/sysdeps/linux/e1/bits/kernel_types.h
--- uClibc-0.9.26-orig/libc/sysdeps/linux/e1/bits/kernel_types.h	2003-10-08 20:15:46.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/e1/bits/kernel_types.h	2004-02-06 08:07:07.000000000 +0100
@@ -30,6 +30,7 @@
 typedef unsigned int	__kernel_gid32_t;
 typedef unsigned short	__kernel_old_uid_t;
 typedef unsigned short	__kernel_old_gid_t;
+typedef __kernel_dev_t	__kernel_old_dev_t;
 typedef long long	__kernel_loff_t;
 
 /*
diff -Naur uClibc-0.9.26-orig/libc/sysdeps/linux/i386/bits/kernel_types.h uClibc-0.9.26/libc/sysdeps/linux/i386/bits/kernel_types.h
--- uClibc-0.9.26-orig/libc/sysdeps/linux/i386/bits/kernel_types.h	2002-08-27 03:20:12.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/i386/bits/kernel_types.h	2004-02-06 08:07:08.000000000 +0100
@@ -30,6 +30,7 @@
 typedef unsigned int	__kernel_gid32_t;
 typedef unsigned short	__kernel_old_uid_t;
 typedef unsigned short	__kernel_old_gid_t;
+typedef __kernel_dev_t	__kernel_old_dev_t;
 typedef long long	__kernel_loff_t;
 
 typedef struct {
diff -Naur uClibc-0.9.26-orig/libc/sysdeps/linux/m68k/bits/kernel_types.h uClibc-0.9.26/libc/sysdeps/linux/m68k/bits/kernel_types.h
--- uClibc-0.9.26-orig/libc/sysdeps/linux/m68k/bits/kernel_types.h	2002-08-27 03:20:13.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/m68k/bits/kernel_types.h	2004-02-06 08:07:09.000000000 +0100
@@ -31,6 +31,7 @@
 typedef unsigned short	__kernel_old_uid_t;
 typedef unsigned short	__kernel_old_gid_t;
 typedef long long	__kernel_loff_t;
+typedef __kernel_dev_t	__kernel_old_dev_t;
 
 typedef struct {
 #ifdef __USE_ALL
diff -Naur uClibc-0.9.26-orig/libc/sysdeps/linux/microblaze/bits/kernel_types.h uClibc-0.9.26/libc/sysdeps/linux/microblaze/bits/kernel_types.h
--- uClibc-0.9.26-orig/libc/sysdeps/linux/microblaze/bits/kernel_types.h	2003-10-14 13:52:29.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/microblaze/bits/kernel_types.h	2004-02-06 08:07:10.000000000 +0100
@@ -42,6 +42,7 @@
 
 typedef unsigned short	__kernel_old_uid_t;
 typedef unsigned short	__kernel_old_gid_t;
+typedef __kernel_dev_t	__kernel_old_dev_t;
 
 typedef struct {
 #ifdef __USE_ALL
diff -Naur uClibc-0.9.26-orig/libc/sysdeps/linux/mips/bits/kernel_types.h uClibc-0.9.26/libc/sysdeps/linux/mips/bits/kernel_types.h
--- uClibc-0.9.26-orig/libc/sysdeps/linux/mips/bits/kernel_types.h	2003-02-23 06:39:26.000000000 +0100
+++ uClibc-0.9.26/libc/sysdeps/linux/mips/bits/kernel_types.h	2004-02-06 08:07:11.000000000 +0100
@@ -31,6 +31,7 @@
 typedef int		__kernel_gid32_t;
 typedef __kernel_uid_t	__kernel_old_uid_t;
 typedef __kernel_gid_t	__kernel_old_gid_t;
+typedef __kernel_dev_t	__kernel_old_dev_t;
 typedef long long      __kernel_loff_t;
 #else
 typedef unsigned int	__kernel_dev_t;
@@ -66,6 +67,7 @@
 typedef int		__kernel_gid32_t;
 typedef __kernel_uid_t	__kernel_old_uid_t;
 typedef __kernel_gid_t	__kernel_old_gid_t;
+typedef __kernel_dev_t	__kernel_old_dev_t;
 typedef long long      __kernel_loff_t;
 #endif

diff -Naur uClibc-0.9.26-orig/libc/sysdeps/linux/powerpc/bits/kernel_types.h uClibc-0.9.26/libc/sysdeps/linux/powerpc/bits/kernel_types.h
--- uClibc-0.9.26-orig/libc/sysdeps/linux/powerpc/bits/kernel_types.h	2002-08-27 03:20:17.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/powerpc/bits/kernel_types.h	2004-02-06 08:07:12.000000000 +0100
@@ -33,6 +33,7 @@
 typedef unsigned int	__kernel_gid32_t;
 typedef unsigned int	__kernel_old_uid_t;
 typedef unsigned int	__kernel_old_gid_t;
+typedef __kernel_dev_t	__kernel_old_dev_t;
 #else
 typedef unsigned int	__kernel_dev_t;
 typedef unsigned int	__kernel_ino_t;
@@ -57,6 +58,7 @@
 typedef unsigned int	__kernel_gid32_t;
 typedef unsigned int	__kernel_old_uid_t;
 typedef unsigned int	__kernel_old_gid_t;
+typedef __kernel_dev_t	__kernel_old_dev_t;
 typedef long long	__kernel_loff_t;
 #endif
 
diff -Naur uClibc-0.9.26-orig/libc/sysdeps/linux/sh/bits/kernel_types.h uClibc-0.9.26/libc/sysdeps/linux/sh/bits/kernel_types.h
--- uClibc-0.9.26-orig/libc/sysdeps/linux/sh/bits/kernel_types.h	2002-08-27 03:20:18.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/sh/bits/kernel_types.h	2004-02-06 08:07:12.000000000 +0100
@@ -30,6 +30,7 @@
 typedef unsigned int	__kernel_gid32_t;
 typedef unsigned short	__kernel_old_uid_t;
 typedef unsigned short	__kernel_old_gid_t;
+typedef __kernel_dev_t	__kernel_old_dev_t;
 typedef long long	__kernel_loff_t;
 
 typedef struct {
diff -Naur uClibc-0.9.26-orig/libc/sysdeps/linux/sh64/bits/kernel_types.h uClibc-0.9.26/libc/sysdeps/linux/sh64/bits/kernel_types.h
--- uClibc-0.9.26-orig/libc/sysdeps/linux/sh64/bits/kernel_types.h	2003-08-13 22:08:29.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/sh64/bits/kernel_types.h	2004-02-06 08:07:14.000000000 +0100
@@ -42,6 +42,7 @@
 typedef unsigned int    __kernel_gid32_t;
 typedef unsigned short  __kernel_old_uid_t;
 typedef unsigned short  __kernel_old_gid_t;
+typedef __kernel_dev_t	__kernel_old_dev_t;
 typedef long long       __kernel_loff_t;
 
 typedef struct {
diff -Naur uClibc-0.9.26-orig/libc/sysdeps/linux/sparc/bits/kernel_types.h uClibc-0.9.26/libc/sysdeps/linux/sparc/bits/kernel_types.h
--- uClibc-0.9.26-orig/libc/sysdeps/linux/sparc/bits/kernel_types.h	2002-08-27 03:20:20.000000000 +0200
+++ uClibc-0.9.26/libc/sysdeps/linux/sparc/bits/kernel_types.h	2004-02-06 08:07:16.000000000 +0100
@@ -30,6 +30,7 @@
 typedef unsigned short	       __kernel_gid16_t;
 typedef __kernel_uid_t 	       __kernel_old_uid_t;
 typedef __kernel_gid_t         __kernel_old_gid_t;
+typedef __kernel_dev_t         __kernel_old_dev_t;
 typedef __kernel_uid_t	       __kernel_uid32_t;
 typedef __kernel_gid_t	       __kernel_gid32_t;
 typedef int		       __kernel_suseconds_t;
@@ -59,6 +60,7 @@
 typedef unsigned int	       __kernel_gid32_t;
 typedef unsigned short	       __kernel_old_uid_t;
 typedef unsigned short	       __kernel_old_gid_t;
+typedef __kernel_dev_t         __kernel_old_dev_t;
 typedef long long              __kernel_loff_t;
 #endif
 
diff -Naur uClibc-0.9.26-orig/libc/sysdeps/linux/v850/bits/kernel_types.h uClibc-0.9.26/libc/sysdeps/linux/v850/bits/kernel_types.h
--- uClibc-0.9.26-orig/libc/sysdeps/linux/v850/bits/kernel_types.h	2003-01-31 10:20:24.000000000 +0100
+++ uClibc-0.9.26/libc/sysdeps/linux/v850/bits/kernel_types.h	2004-02-06 08:07:16.000000000 +0100
@@ -40,6 +40,7 @@
 
 typedef unsigned short	__kernel_old_uid_t;
 typedef unsigned short	__kernel_old_gid_t;
+typedef __kernel_dev_t	__kernel_old_dev_t;
 
 typedef struct {
 #ifdef __USE_ALL
diff -Naur uClibc-0.9.26-orig/libc/sysdeps/linux/h8300/bits/kernel_types.h uClibc-0.9.26/libc/sysdeps/linux/h8300/bits/kernel_types.h
--- uClibc-0.9.26-orig/libc/sysdeps/linux/h8300/bits/kernel_types.h	2002-11-21 07:43:15.000000000 +0100
+++ uClibc-0.9.26/libc/sysdeps/linux/h8300/bits/kernel_types.h	2004-07-15 09:33:59.000000000 +0200
@@ -1,11 +1,11 @@
-/* Note that we use the exact same include guard #define names
- * as asm/posix_types.h.  This will avoid gratuitous conflicts 
- * with the posix_types.h kernel header, and will ensure that 
- * our private content, and not the kernel header, will win.
- *  -Erik
- */
-#ifndef __ARCH_H8300_POSIX_TYPES_H
-#define __ARCH_H8300_POSIX_TYPES_H
+#ifndef _BITS_KERNEL_TYPES_H
+#define _BITS_KERNEL_TYPES_H
+
+/* Sigh.  We need to carefully wrap this one...  No guarantees
+ * that the asm/posix_types.h kernel header is working.  Many
+ * arches have broken headers that introduce tons of gratuitous
+ * conflicts with uClibc's namespace.   See bits/kernel_types.h
+ * for i386, arm, etc for examples... */
 
 typedef unsigned short	__kernel_dev_t;
 typedef unsigned long	__kernel_ino_t;
@@ -28,13 +28,10 @@
 typedef unsigned short	__kernel_gid16_t;
 typedef unsigned int	__kernel_uid32_t;
 typedef unsigned int	__kernel_gid32_t;
-
 typedef unsigned short	__kernel_old_uid_t;
 typedef unsigned short	__kernel_old_gid_t;
-
-#ifdef __GNUC__
 typedef long long	__kernel_loff_t;
-#endif
+typedef __kernel_dev_t	__kernel_old_dev_t;
 
 typedef struct {
 #ifdef __USE_ALL
@@ -44,4 +41,4 @@
 #endif
 } __kernel_fsid_t;
 
-#endif /* __ARCH_H8300_POSIX_TYPES_H */
+#endif /* _BITS_KERNEL_TYPES_H */


